webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/MockServices.scala

```diff
@@ -0,0 +1,1254 @@
+import com.normation.rudder.repository.RoRuleRepository
+import com.normation.rudder.repository.WoRuleRepository
+import com.normation.rudder.services.policies.NodeConfiguration
+import com.normation.rudder.services.policies.ParameterForConfiguration
+import com.normation.rudder.services.policies.Policy
+import org.joda.time.format.ISODateTimeFormat
+
+import scala.collection.SortedMap
+
+/*
+ * Mock services for test, especially reposositories, and provides
+ * test data (nodes, directives, etc)
+ */
+
+object NoTags {
+  def apply() = Tags(Set())
+}
+
+object MkTags {
+  def apply(tags: (String, String)*) = {
+    Tags(tags.map{ case (k, v) => Tag(TagName(k), TagValue(v))}.toSet)
+  }
+}
+
+class MockGitConfigRepo(prefixTestResources: String = "") {
+
+  /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+  // set up root node configuration
+  /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+  val t0 = System.currentTimeMillis()
+
+  val abstractRoot = new File("/tmp/test-rudder-config-repo-" + DateTime.now.toString())
+  abstractRoot.mkdirs()
+  if(System.getProperty("tests.clean.tmp") != "false") {
+    java.lang.Runtime.getRuntime.addShutdownHook(new Thread(new Runnable {
+      override def run(): Unit = FileUtils.deleteDirectory(abstractRoot)
+    }))
+  }
+
+  // config-repo will also be the git root, as a normal rudder
+  val configurationRepositoryRoot = new File(abstractRoot, "configuration-repository")
+  //initialize config-repo content from our test/resources source
+
+  FileUtils.copyDirectory( new File(prefixTestResources + "src/test/resources/configuration-repository") , configurationRepositoryRoot)
+
+  val t1 = System.currentTimeMillis()
+  println(s"Paths inits             : ${t1-t0} ms")
+
+  val gitRepo = GitRepositoryProviderImpl.make(configurationRepositoryRoot.getAbsolutePath).runNow
+  val t2      = System.currentTimeMillis()
+  println(s"Git repo provider       : ${t2-t1} ms")
+}
+
+object MockTechniques {
+  def apply(mockGitConfigRepo: MockGitConfigRepo) = new MockTechniques(mockGitConfigRepo.configurationRepositoryRoot, mockGitConfigRepo.gitRepo)
+}
+
+class MockTechniques(configurationRepositoryRoot: File, gitRepo: GitRepositoryProvider) {
+  val t2      = System.currentTimeMillis()
+  val variableSpecParser = new VariableSpecParser
+  val t2bis = System.currentTimeMillis()
+  println(s"var Spec Parser        : ${t2bis-t2} ms")
+  val systemVariableServiceSpec = new SystemVariableSpecServiceImpl()
+  val t3 = System.currentTimeMillis()
+  println(s"System Var Spec service : ${t3-t2bis} ms")
+
+  val techniqueParser: TechniqueParser = new TechniqueParser(
+      variableSpecParser
+    , new SectionSpecParser(variableSpecParser)
+    , systemVariableServiceSpec
+  )
+  val t4 = System.currentTimeMillis()
+  println(s"Technique parser        : ${t4-t3} ms")
```

Comment:
Do you want to keep println here ? 

I have no idea, it's in test it does no harm